app/components/avo/tab_switcher_component.rb

```diff
@@ -63,36 +63,31 @@ def selected?(tab)
   # Because the developer hasn't specified that it should be visible on edit views (with the show_on: :edit option),
   # the field should not be visible in the item switcher either.
   def visible_items
-    tabs.select do |item|
-      visible = true
+    tabs.select do |tab|
+      next false if tab.items.blank?
+      next false if tab.is_field? && !tab.authorized?
 
-      if item.items.blank?
-        visible = false
+      if tab.items.count == 1
+        next false if !single_item_visible?(tab.items.first)
```

Comment:
can't these be combined into one line?

`tab.items.count == 1 && !single_item_visible?(tab.items.first)`

Reply:
maybe we coudl add a `has_a_single_item?` method on the tab and make it more readable here?